test/ess_rhat.jl

```diff
@@ -32,7 +32,138 @@ function LogDensityProblems.capabilities(p::CauchyProblem)
     return LogDensityProblems.LogDensityOrder{1}()
 end
 
-@testset "ess.jl" begin
+mymean(x) = mean(x)
+
+@testset "ess_rhat.jl" begin
+    @testset "ess/ess_rhat/rhat basics" begin
+        @testset "only promote eltype when necessary" begin
+            @testset for kind in [:rank, :bulk, :tail, :basic]
```

Comment:
Maybe use a tuple instead of a vector (also below)?
```suggestion
            @testset for kind in (:rank, :bulk, :tail, :basic)
```